packages/astro/src/runtime/server/index.ts

```diff
@@ -433,15 +419,34 @@ export async function renderEndpoint(mod: EndpointHandler, params: any) {
 	return body;
 }
 
+// Calls a component and renders it into a string of HTML
+export async function renderToString(result: SSRResult, componentFactory: AstroComponentFactory, props: any, children: any) {
+	const Component = await componentFactory(result, props, children);
+	let template = await renderAstroComponent(Component);
+
+	// <!--astro:head--> injected by compiler
+	// Must be handled at the end of the rendering process
+	if (template.indexOf('<!--astro:head-->') > -1) {
+		template = template.replace('<!--astro:head-->', await renderHead(result));
+	}
```

Comment:
`<!--astro:head-->` is injected by the compiler. If we see this, it's at the end of the `<head>` element and we should inject the links/styles/scripts.